sdk/servicebus/service-bus/src/core/messageReceiver.ts

```diff
@@ -973,6 +972,9 @@ export class MessageReceiver extends LinkEntity {
         );
         try {
           this._onError(err);
+          // Once the user's error handler has been called,
+          // close the receiver to prevent future messages/errors from being received.
+          await this.close();
         } catch (err) {
           log.error(
             "[%s] User-code error in error handler called after disconnect: %O",
```

Comment:
This log message is not totally accurate now since the error could be from `close()`. 

Maybe should be separate try/catch's?

Reply:
It's more inaccurate than that even, now that it can also be called by a non-retryable error from a connection disconnect.

Reply:
Moved `close` out of this try/catch.